File: backend/app/schemas/wire.py
"""Pydantic schemas for wire transfers."""
from pydantic import BaseModel, Field, condecimal
from datetime import datetime
from typing import Optional
from decimal import Decimal


class WireBase(BaseModel):
    """Base wire schema."""

    sender_name: str = Field(..., min_length=1, max_length=200)
    recipient_name: str = Field(..., min_length=1, max_length=200)
    amount: condecimal(max_digits=15, decimal_places=2, gt=0)  # type: ignore
    currency: str = Field(default="USD", pattern="^[A-Z]{3}$")


class WireCreate(WireBase):
    """Schema for creating a wire transfer."""

    pass


class WireUpdate(BaseModel):
    """Schema for updating a wire transfer."""

    sender_name: Optional[str] = Field(None, min_length=1, max_length=200)
    recipient_name: Optional[str] = Field(None, min_length=1, max_length=200)
    amount: Optional[condecimal(max_digits=15, decimal_places=2, gt=0)] = None  # type: ignore
    currency: Optional[str] = Field(None, pattern="^[A-Z]{3}$")
    status: Optional[str] = Field(None, pattern="^(pending|processing|completed|failed)$")


class WireResponse(WireBase):
    """Schema for wire transfer response."""

    id: int
    status: str
    reference_number: Optional[str]
    created_by: int
    created_at: datetime
    updated_at: Optional[datetime]

    class Config:
        from_attributes = True


class WireListResponse(BaseModel):
    """Schema for paginated wire list response."""

    wires: list[WireResponse]
    total: int
    page: int
    page_size: int
    cached: bool = False

File: frontend/src/types/wire.ts
/**
 * Wire transfer type definitions
 */

export enum WireStatus {
  PENDING = 'pending',
  PROCESSING = 'processing',
  COMPLETED = 'completed',
  FAILED = 'failed',
}

export interface Wire {
  id: number;
  sender_name: string;
  recipient_name: string;
  amount: number;
  currency: string;
  status: WireStatus;
  reference_number: string;
  created_by: number;
  created_at: string;
  updated_at: string | null;
}

export interface WireCreate {
  sender_name: string;
  recipient_name: string;
  amount: number;
  currency: string;
}

export interface WireUpdate {
  sender_name?: string;
  recipient_name?: string;
  amount?: number;
  currency?: string;
  status?: WireStatus;
}

export interface WireListResponse {
  wires: Wire[];
  total: number;
  page: number;
  page_size: number;
  cached: boolean;
}

export interface User {
  id: number;
  email: string;
  is_active: boolean;
  created_at: string;
}

export interface LoginCredentials {
  email: string;
  password: string;
}

export interface RegisterData {
  email: string;
  password: string;
}

export interface AuthTokens {
  access_token: string;
  refresh_token: string;
  token_type: string;
}
